app/services/SMService.scala

```diff
@@ -93,14 +92,23 @@ trait SMService extends Logging {
     jsonConnector.loadServicesJson.fields map { case (service, _) => service}
   }
 
-  def searchForService(query: String): Seq[String]= {
-    if(query.contains("\"")) {
-      exactMatchSearch(jsonConnector.loadServicesJson.fields, query)
-    } else {
-      jsonConnector.loadServicesJson.fields.collect { case (name, _) if name.contains(query) => name}
+  def searchForService(query: String): Seq[String] = {
+    val servicesJson: Seq[(String, JsValue)] = jsonConnector.loadServicesJson.fields
+
+    (isAnInt(query), query.contains("\"")) match {
+      case (true, _) => searchServicesUsingPort(query, servicesJson)
+      case (_, true) => exactMatchSearch(servicesJson, query.toUpperCase)
+      case _ => servicesJson.collect { case (name, _) if name.contains(query.toUpperCase) => name }
```

Comment:
servicesJson.filter(_._1.contains(query.toUpperCase) could work aswell

Reply:
Yep, that would work and would return more data than we are currently looking for (Seq[String] vs Seq[(String, JsValue)]). However, in the context of how searchForService is used, we only need the service name. The only change to that line from before was to transform the query to uppercase.